zds/member/forms.py

```diff
@@ -508,24 +422,47 @@ def clean(self):
         # Clean data
         username = cleaned_data.get('username')
         email = cleaned_data.get('email')
+        print(username, email)
 
-        # Check that the username or the email is filled
-        if (username and email) or (not username and not email):
-            if username and email:
-                self._errors['username'] = self.error_class([_(u'Les deux champs ne doivent pas être rempli. '
-                                                               u'Remplissez soit l\'adresse de courriel soit le '
-                                                               u'nom d\'utilisateur')])
-            else:
-                self._errors['username'] = self.error_class([_(u'Il vous faut remplir au moins un des deux champs')])
+        if username and email:
+            self._errors['username'] = self.error_class([_(u'Les deux champs ne doivent pas être rempli. Remplissez soi'
+                                                           u't l\'adresse de courriel soit le nom d\'utilisateur')])
+        elif not username and not email:
+            self._errors['username'] = self.error_class([_(u'Il vous faut remplir au moins un des deux champs')])
         else:
-            # Check if the user exist
+            # run validators
             if username:
-                if User.objects.filter(Q(username=username)).count() == 0:
-                    self._errors['username'] = self.error_class([_(u'Ce nom d\'utilisateur n\'existe pas')])
+                validate_not_empty(username)
```

Comment:
`if username is not None:`

Sinon ça sert à rien de vérifier que c'est pas `empty` ;)